doc/build.md

```diff
@@ -44,6 +44,39 @@ If not, you should review carefully the error messages and [ask for help](http:/
 
 Note that if you're connecting to the internet through proxies, you need to [tell maven how to reach public repositories](http://maven.apache.org/guides/mini/guide-proxies.html).
 
+## GeoNetwork
+
+### Important note about the versions
+
+As of december 2015, the default version for GeoNetwork in geOrchestra switched
+to the 3.0.4 version. But some existing instances are still relying on features
+that are present in the older versions, hence it is still possible to use the
+GeoNetwork 2.x versions in geOrchestra.
+
+
+If you need a 2.x version, you will need to checkout the appropriate version in
+your repository before launching the build in the next step.
+
+Assuming a GN2 version is needed and the targeted version of geOrchestra is
+the `15.12` one, follow these steps before compiling:
+
+```
+cd geonetwork
+git checkout georchestra-15.12
+```
+
+This step would have to be repeated each time the git submodules are updated
+(using one of these commands `git submodule sync`, `git submodule update` ...).
+
+
+### Migrating from GeoNetwork 2 to 3.0.4
+
+Currently, no method is provided to migrate an older georchestra-geonetwork
+database to the new version, you will have to export your data from the old
+database by hand, start from an empty Geonetwork3 database, then reimport your
+data (using a batch insert for instance).
```

Comment:
Difficult in a gist, because it is mainly UI interactions. As discussed with @landryb the automatic upgrade process deserve another try, since some bugs have been fixed since last realign.